sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -466,7 +469,7 @@ public String toString() {
   @Experimental(Experimental.Kind.SOURCE_SINK)
   @AutoValue
   public abstract static class Write
-      extends PTransform<PCollection<KV<ByteString, Iterable<Mutation>>>, PDone> {
+      extends PTransform<PCollection<KV<ByteString, Iterable<Mutation>>>, PCollection<Void>> {
```

Comment:
Just a heads up, this is a backward incompatible change, it will break people's code of the form:

```
public class MyTransform extends PTransform<..., PDone> {
  expand {
    return blah.apply(BigtableIO.write());
  }
}
```

Up to you whether this is a big concern or no. If you want to make it compatible, follow the example of TextIO.write().withOutputFilenames() (or something like that).

Reply:
the class is marked as experimental, which means it can have backwards-incompatible changes right? ;)

I'd be ok with either way, but if we're going by the letter of the `@Experimental` rules, it seems like it'd be cleaner to just change the interface rather than adding a new option.

Reply:
Thanks @jkff for pointing out that this is backwards incompatible. @steveniemitz even though this is marked experimental there are many of users who use this connector. So it's preferable if we can not break user pipelines (which can be easily achieved with a slight change here).

Reply:
hah, I was mostly just teasing since like 50% of beam is marked as experimental at this point.

What's the path forward here for merging this in, if I make this change are we good to go?

Reply:
should I just remove the @Experimental annotation and comment while I'm in here?  If we're saying that we can't make backwards-incompatible changes to the transform, I'm not sure what use having that in here still is.

Reply:
Yeah, I think it makes sense to remove the experimental annotation from BigTable connector since it has been around for some time.

Beta reference should definitely be removed :)

cc: @aaltay @sduskis

Reply:
Is really a PR that is literally changing the current API the best place to decide that we should remove the `Experimental` characteristic of the IO? I mean the APIs have been very stable but this decision is probably worth of a different issue, and given the uncertain future of Writes with return type I think probably it is wise to let it at least for the Write signature for some time.

Reply:
Also with the ongoing work on [BEAM-7615](https://issues.apache.org/jira/browse/BEAM-7615) I am not even sure the API will stay stable still so definitely -1 on any API stability change at least for this PR.

Reply:
thats fine with me...I just want to land this at this point :P

Reply:
I think we can remove the experimental tag from stable transforms (Read, Write). This PR does not break backwards compatibility and I don't think we should break backwards compatibility for these transforms in future PRs either. This does not prevent adding new features or adding a new "ReadAll" form transform when we have SDF.

I agree that this is better done in a separate JIRA/PR. Created https://issues.apache.org/jira/browse/BEAM-7631.

Reply:
You are right about backwards compatibility @chamikaramj, I had in mind that we were still changing the return time so it is ok. `BigtableIO.ReadAll` may require refactors, but it can indeed be still backwards compatible as we did for `HBaseIO.ReadAll`. So probably good to go then in the other issue. I think we can do that for HBaseIO too since it has barely changed in 2 years (apart of the ReadAll addition.

Reply:
We should probably mark `BigtableIO.ReadAll` experimental for a bit of time. Just created https://issues.apache.org/jira/browse/BEAM-7633 for the HBaseIO case.

Reply:
If we go with this change, how about returning PCollection<?>. This means we will allow ourselves, in a forwards compatible way, to return something other than Void (e.g. some kind of a status about what was written) in the future.

Reply:
@robertwb may have a point but it `?` have other trade-offs too. I am going to open a discussion thread on the ML around the return types of Write transforms to maybe have other opinions and to not fill this PR with the pros/cons of each approach.

Reply:
Email [sent](https://lists.apache.org/thread.html/d1a4556a1e13a661cce19021926a5d0997fbbfde016d36989cf75a07@%3Cdev.beam.apache.org%3E).
@robertwb @chamikaramj does python already have this Write with return pattern in some IOs? Is this an issue there? Or the lack of types makes things ‘easier’?

Reply:
Thanks for sending the email. Replied there.